src/index.ts

```diff
@@ -133,13 +131,18 @@ function HalJsonVuex (store: Store<Record<string, State>>, axios: AxiosInstance,
    * @returns Promise   Resolves when the GET request has completed and the updated entity is available
    *                    in the Vuex store.
    */
-  async function reload (uriOrEntity: string | Resource | StoreData | EmbeddedCollectionMeta): Promise<Resource | EmbeddedCollection> {
-    // For embedded collections which have no self link, reload the parent entity instead
-    const uri = normalizeEntityUri(isEmbeddedCollection(uriOrEntity) ? uriOrEntity._meta.reload.uri : uriOrEntity, axios.defaults.baseURL)
+  async function reload (uriOrEntity: string | ResourceInterface): Promise<ResourceInterface> {
+    if (uriOrEntity instanceof Resource && 'virtual' in uriOrEntity._storeData._meta && uriOrEntity._storeData._meta.virtual) {
+      // For embedded collections which had to reload the parent entity, unwrap the embedded collection after loading has finished
+      const { owningResource, owningRelation } = uriOrEntity._storeData._meta
+      return reload(owningResource).then(owner => owner[owningRelation]())
+    }
```

Comment:
I don't really like that reloading a virtual resource only works if I am passing a `Resource` into `reload`, but not with a (virtual) URI. Consider moving this block down to line 151, and then checking if `get(uriOrEntity)` is virtual (maybe check first with `isUnknown` so we don't accidentally try to fetch a virtual URI from the API). This way, we could also get rid of the whole `reloadResourceFromStoreData` function i think: In `deleted` (the only place that actually uses `reloadResourceFromStoreData`) we could just extract the `self` link from the `StoreData` and use the refactored `reload`.

I think you were trying to eliminate `StoreData` from the possible types for `uriOrEntity` here. I would support that idea, and you should remove `StoreData` from the accepted argument types of `normalizeEntityUri` as well.

Reply:
Like the idea. Implemented in https://github.com/ecamp/hal-json-vuex/pull/260/commits/ad6ec9dddafa75c5ecc495da109dcb9ea49a64b8

Checking `isUnknown` is unnecessary in my opinion: A virtual resource cannot really be unknown by definition, because it has previously been loaded & stored via another resource. If I manually present a virtual URI (or any other random URI for that sake), which `hal-json-vuex` has not seen before or has been purged in the meantime, then yes, the loading will fail. But that's more like a usage error to me: In this scenario, there's no way `hal-json-vuex` can tell, the URI is virtual, unless we start to parse the URI string (which I believe we shouldn't).

Reply:
Okay, but then we should disallow purging virtual uris from the store

Reply:
Isn't this the user's responsibility: Don't purge a virtual resource & then immediately try to reload it afterwards.

Similar as: Don't delete an entity & then try to reload it afterwards.

Speaking of deletion: At the moment we also don't limit `del`, `patch` and `post` on the top level. Within the Resource class, this is prohibited. But it's not double-checked again in the actual function.

So this would throw an error
```javascript
virtualResource.$del() // $del is not implemented for virtual resources
```
but this would try to perform a server call:
```
api.del(virtualResource)
```

Do we want to prohibit this as well?

Reply:
Ah, yes, good point. I think it would be better to move these exceptions from the `$del` etc. versions to the `api.del` versions. This way it will behave consistently, independently of which paradigm one uses.

I think after that, we can merge this.

Reply:
> Ah, yes, good point. I think it would be better to move these exceptions from the `$del` etc. versions to the `api.del` versions. This way it will behave consistently, independently of which paradigm one uses.

Yes, agree. Better to have the check in the main `api.del`. I won't be able to progress on this before next weekend. If you fancy to work on it, feel free to go ahead. Otherwise I'll finalize during or after the hackathon.